src/bosh-director/lib/bosh/blobstore_client/gcscli_blobstore_client.rb

```diff
@@ -0,0 +1,125 @@
+require 'securerandom'
+require 'open3'
+require 'json'
+
+module Bosh::Blobstore
+  class GcscliBlobstoreClient < BaseClient
+
+    EXIT_CODE_OBJECT_NOTFOUND = 3
+
+    # Blobstore client for GCS, using bosh-gcscli binary
+    # @param [Hash] options GCSconnection options
+    # @option options [Symbol] bucket_name
+    #   key that is applied before the object is sent to GCS
+    # @option options [Symbol, optional] access_key_id
+    # @option options [Symbol, optional] secret_access_key
+    # @option options [Symbol] gcscli_path
+    #   path to gcscli binary
+    # @option options [Symbol, optional] gcscli_config_path
+    #   path to store configuration files
+    # @note If access_key_id and secret_access_key are not present, the
+    #   blobstore client operates in read only mode
+    def initialize(options)
+      super(options)
+
+      @gcscli_path = @options.fetch(:gcscli_path)
+      unless Kernel.system("#{@gcscli_path}", "--v", out: "/dev/null", err: "/dev/null")
+        raise BlobstoreError, "Cannot find gcscli executable. Please specify gcscli_path parameter"
+      end
+
+      @gcscli_options = {
+        bucket_name: @options[:bucket_name],
+        credentials_source: @options.fetch(:credentials_source, 'none'),
```

Comment:
Read-only tests are included in an additional, smaller PR into this fork. Given that its a 61 line addition, I can include them here if you think that's a reasonable amount to add.